### DISCUSSION TITLE
[Q/A][Application-Level Service Discovery] Dynamic specification of one service at runtime when configure "provided-by=A,B,C"

### DISCUSSION THREAD

**Discussion by dgqypl** [apache/dubbo#13983] [Question] [answered]
### Pre-check

- [X] I am sure that all the content I provide is in English.

### Apache Dubbo Component

Java SDK (apache/dubbo)

### Details

In Dubbo 3, when using Application-Level Service Discovery, for example, when I configure this snippet in my consumer yaml:
```yaml
  ...
  consumer:
    provided-by: A,B,C
```

I want to achieve a purpose that I can specify the concrete service when invoke RPC call each time, such as I specify service C when invoke RPC call this time, and I specify service A when invoke RPC call for the next time.

Is there anyway to achieve my purpose? thx.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by AlbumenJ** [comment, marked as answer]
This is a good question. We don't have an elegant way to specify `provided by` in a large app.
